### ISSUE TITLE
[BUG] Harmony Hub won’t show up

### ISSUE THREAD

**Issue by mbac** [nicoduj/homebridge-harmony#379]
Hi,

Despite apparently successful setup, accessories presented via my Harmony Hub won’t show up in the Home app…

The Harmony Hub appears correctly between other hubs in the home, but there are no activities or TV. At startup, Homebridge doesn’t report any error. The last few lines are:

```
Preparing Advertiser for 'Harmony Hub-TV 18C2' using bonjour-hap backend!
Starting to advertise 'Harmony Hub-TV 18C2' using bonjour-hap backend!
[10/9/2021, 11:59:30 AM] Harmony Hub-TV 18C2 is running on port 51254.
[10/9/2021, 11:59:30 AM] Please add [Harmony Hub-TV 18C2] manually in Home app. Setup Code: 287-23-854
```

The setup code is the same as the main Homebridge hub–is this expected? When I enter the code manually, the procedure times out.

Can you please help me troubleshoot?

**Comment by nicoduj** [comment]
Could you please share your config ? You indeed have to add it manually through the code provided yes , but i don't understand what you mean by other hubs ?But you should be able to discover it also .

**Comment by mbac** [comment]
Hi and thanks for your reply. By “other hubs” I actually meant bridges: Philips Hue bridge, a Netatmo bridge, the Homebridge… Sorry about that. The Harmony hub does show up in the list of bridges available in the Home app, but no activity is shown in the relevant room.

Current config is:
```
{
            "platform": "HarmonyHubWebSocket",
            "name": "Harmony Hub",
            "hubIP": "192.168.3.251",
            "hubName": "Harmony Hub",
            "DELAY_BEFORE_RETRY_AFTER_NETWORK_LOSS": 60000,
            "HUB_CONNECT_TIMEOUT": 10000,
            "HUB_SEND_TIMEOUT": 30000,
            "cleanCache": true,
            "TVAccessory": true,
            "mainActivity": "Sky",
            "playPauseBehavior": true
        }
```

**Comment by nicoduj** [comment]
Ok, thx.
first of all, remove cleanCacheOption.
Then, did you add homebridge itself to home app first ? Your config will only add one TVAccessory in home app, but you have to add the bridge and the TV accessory.

**Comment by mbac** [comment]
Yes, Homebridge was added first and there are several other accessories connected through it which work fine.

Could you please explain what you mean by “add the bridge and the TV accessory”?

Thanks!

**Comment by nicoduj** [comment]
Hi, 

sorry for my late answer. 

If you have other bridged accssories available, it means you have added the homebridge hub itself. The TV accessory is exposed outside the hub, on its own port. Are you sure this dedicated port is available / open ? From your log, it seems to be 51254 . You can specifiy in homebridge conf the port range to be used for external accessories like the TV accessory the plugin exposes.

bonjour backend have some times troubles on some conf, you can try ciao backend to see if it is better on your conf.

**Comment by atlienk** [comment]
Has a resolution to this issue been identified? I believe that I'm encountering something similar and could use some assistance. 

I'm able to add my Harmony Hub into HomeBridge (it will appear as an Accessory), but not into the HomeKit app itself. What's odd is that HomeBridge reports that it cannot connect to the Harmony Hub over port 8088, despite it being open. HomeBridge then reports that it's connecting to the Harmony Hub on a different port. If I try to use the reported code to add the Harmony Hub to HomeKit, nothing happens. I've even gone as far as to remove HomeBridge from HomeKit, and then readd it. All other accessories reconnect without issue except for the Harmony Hub.

Log file export below.

[03/04/2022, 21:45:04] [Harmony Hub] (Harmony Hub)Error - Error retrieving info from hub : request to http://192.168.2.3:8088/ failed, reason: connect EHOSTUNREACH 192.168.2.3:8088-(16)
[03/04/2022, 21:45:14] [Harmony Hub] (Harmony Hub)INFO - Loading activities...
[03/04/2022, 21:45:17] [Harmony Hub] (Harmony Hub)Error - Error retrieving info from hub : request to http://192.168.2.3:8088/ failed, reason: connect EHOSTUNREACH 192.168.2.3:8088-(17)
[03/04/2022, 21:45:27] [Harmony Hub] (Harmony Hub)INFO - Loading activities...
[03/04/2022, 21:45:30] [Harmony Hub] (Harmony Hub)Error - Error retrieving info from hub : request to http://192.168.2.3:8088/ failed, reason: connect EHOSTUNREACH 192.168.2.3:8088-(18)
[03/04/2022, 21:45:40] [Harmony Hub] (Harmony Hub)INFO - Loading activities...
[03/04/2022, 21:45:41] [Harmony Hub] (Harmony Hub)INFO - Adding Accessory : Harmony Hub-TV
[03/04/2022, 21:45:41] [Harmony Hub] (Harmony Hub)INFO - configuring Main TV Service
[03/04/2022, 21:45:41] [Harmony Hub] (Harmony Hub)INFO - Creating TV Service
[03/04/2022, 21:45:41] [Harmony Hub] (Harmony Hub)INFO - Creating Input Service - Listen to Music
[03/04/2022, 21:45:41] [Harmony Hub] (Harmony Hub)INFO - Creating Input Service - Watch Apple TV
[03/04/2022, 21:45:41] [Harmony Hub] (Harmony Hub)INFO - Configuring Main Activity Watch TV
[03/04/2022, 21:45:41] [Harmony Hub] (Harmony Hub)INFO - Creating TV Speaker Service
[03/04/2022, 21:45:41] [Harmony Hub] (Harmony Hub)INFO - Creating Input Service - Watch TV
[03/04/2022, 21:45:41] [Harmony Hub] (Harmony Hub)INFO - Creating Input Service - Listen to Vinyl
[03/04/2022, 21:45:41] [Harmony Hub] (Harmony Hub)INFO - setupFoundAccessories - TV accessory added as external accessory
[03/04/2022, 21:45:41] Harmony Hub-TV 18C2 is running on port 42289.
[03/04/2022, 21:45:41] Please add [Harmony Hub-TV 18C2] manually in Home app. Setup Code: 419-24-644

**Comment by nicoduj** [comment]
Hi, it seems your hub is not reachable from your homekit server. Are you sure you have the correct IP / it is fixed on your router ? Please share your conf maybe there is something wrong in it.

**Comment by atlienk** [comment]
I have a DHCP reservation for the Hub. It hasn't been changed for years. What's even more peculiar is that I'm seeing the accessory defined in HomeBridge, but it simply can't be added to HomeKit.

Sent with [ProtonMail](https://protonmail.com/) secure email.

------- Original Message -------

**Comment by FreddiH** [comment]
Hi,

I have another problem with my Homebridge, the first two Hubs are added correctly, but the third didnt show up.
Did I messed up somewhere?

My configuration:
   {
   	"bridge": {
   		"name": "Homebridge4Harmony",
   		"username": "CC:22:3D:E3:CE:31",
   		"port": 51832,
   		"pin": "031-45-163",
   		"manufacturer": "@nfarina",
   		"model": "Homebridge",
   		"serialNumber": "Not Specified",
   		"firmwareRevision": "0.4.27",
   		"hardwareRevision": "<optional>"

   	},



   	"platforms": [{
   		"platform": "HarmonyHubWebSocket",
   		"name": "Heimkino HarmonyHub",
   		"hubIP": "192.168.0.16",
   		"cleanCache": false,
   		"TVAccessory": true,
   		"mainActivity": "TV",
   		"switchAccessories": false,
   		"publishGeneralMuteSwitch": false,
   		"publishGeneralVolumeSlider": false,
   		"publishGeneralVolumeSwitches": false,
   		"linkVolumeControlToTV": false,
   		"publishSwitchActivitiesAsIndividualAccessories": false,
   		"otherPlatforms": [{
   			"name": "Schlafzimmer HarmonyHub",
   			"hubIP": "192.168.0.17",
   			"TVAccessory": true,
   			"mainActivity": "Apple TV Schlafzimmer",
   			"activitiesToPublishAsInputForTVMode": [
   				"Apple TV Schlafzimmer"
   			],
   			"switchAccessories": false,
   			"publishGeneralMuteSwitch": false,
   			"publishGeneralVolumeSlider": false,
   			"linkVolumeControlToTV": false,
   			"otherPlatforms": [{
   				"name": "Wohnzimmer HarmonyHub",
   				"hubIP": "192.168.0.26",
   				"TVAccessory": true,
   				"mainActivity": "Apple TV 4K",
   				"activitiesToPublishAsInputForTVMode": [
   					"Apple TV 4K"
   				],
   				"switchAccessories": false,
   				"publishGeneralMuteSwitch": false,
   				"publishGeneralVolumeSlider": false,
   				"linkVolumeControlToTV": false
   			}]
   		}]
   	}]
   }

Thank you very much.

**Comment by nicoduj** [comment]
Hi, other platforms is an array, your two hubs should be in it, you must not repeat other platforms inside. 

"otherPlatforms": [
{
		"name": "Schlafzimmer HarmonyHub",
		"hubIP": "192.168.0.17",
		"TVAccessory": true,
		"mainActivity": "Apple TV Schlafzimmer",
		"activitiesToPublishAsInputForTVMode": [
			"Apple TV Schlafzimmer"
		],
		"switchAccessories": false,
		"publishGeneralMuteSwitch": false,
		"publishGeneralVolumeSlider": false,
		"linkVolumeControlToTV": false
},
{
			"name": "Wohnzimmer HarmonyHub",
			"hubIP": "192.168.0.26",
			"TVAccessory": true,
			"mainActivity": "Apple TV 4K",
			"activitiesToPublishAsInputForTVMode": [
				"Apple TV 4K"
			],
			"switchAccessories": false,
			"publishGeneralMuteSwitch": false,
			"publishGeneralVolumeSlider": false,
			"linkVolumeControlToTV": false
}
]

**Comment by FreddiH** [comment]
thank you very much for your help.
After the change I could not add the third Hub.
But after deleting the Docker container and a clean reinstall, now everything works fine.

If anybody needs my finished config, here is mine:

   {
   	"bridge": {
   		"name": "Homebridge4Harmony",
   		"username": "CC:22:3E:E3:CE:31",
        "start": 52100,
        "end": 52150,
   		"pin": "031-45-165",
   		"manufacturer": "@nfarina",
   		"model": "Homebridge",
   		"serialNumber": "Not Specified",
   		"firmwareRevision": "0.4.27",
   		"hardwareRevision": "<optional>"

   	},



 "platforms": [
  {
    "platform": "HarmonyHubWebSocket",
   		"name": "Heimkino Harmony",
   		"hubIP": "192.168.0.16",
   		"cleanCache": false,
   		"TVAccessory": true,
   		"mainActivity": "Apple TV Heimkino",
   		"switchAccessories": false,
   		"publishGeneralMuteSwitch": false,
   		"publishGeneralVolumeSlider": false,
   		"publishGeneralVolumeSwitches": false,
   		"linkVolumeControlToTV": false,
   		"publishSwitchActivitiesAsIndividualAccessories": false,
   		"otherPlatforms": [{
			"name": "Schlafzimmer HarmonyHub",
			"hubIP": "192.168.0.17",
			"TVAccessory": true,
			"mainActivity": "Apple TV Schlafzimmer",
			"activitiesToPublishAsInputForTVMode": [
				"Apple TV Schlafzimmer"
			],
			"switchAccessories": false,
			"publishGeneralMuteSwitch": false,
			"publishGeneralVolumeSlider": false,
			"linkVolumeControlToTV": false
},
{
			"name": "Wohnzimmer HarmonyHub",
			"hubIP": "192.168.0.26",
			"TVAccessory": true,
			"mainActivity": "Apple TV 4K",
			"activitiesToPublishAsInputForTVMode": [
				"Apple TV 4K"
			],
			"switchAccessories": false,
			"publishGeneralMuteSwitch": false,
			"publishGeneralVolumeSlider": false,
			"linkVolumeControlToTV": false
   			
   		}]
   	}]
   }

**Comment by scottleestrange** [comment]
Other Platforms is an array of other platforms
	"otherPlatforms": [{
		"name": "Schlafzimmer HarmonyHub",
		"hubIP": "192.168.0.17",
		"TVAccessory": true,
		"mainActivity": "Apple TV Schlafzimmer",
		"activitiesToPublishAsInputForTVMode": [
			"Apple TV Schlafzimmer"
		],
		"switchAccessories": false,
		"publishGeneralMuteSwitch": false,
		"publishGeneralVolumeSlider": false,
		"linkVolumeControlToTV": false
		},
		{
			"name": "Wohnzimmer HarmonyHub",
			"hubIP": "192.168.0.26",
			"TVAccessory": true,
			"mainActivity": "Apple TV 4K",
			"activitiesToPublishAsInputForTVMode": [
				"Apple TV 4K"
			],
			"switchAccessories": false,
			"publishGeneralMuteSwitch": false,
			"publishGeneralVolumeSlider": false,
			"linkVolumeControlToTV": false
		}
	]

**Comment by grelca** [comment]
I'm having this issue running HB on MacOS Monterey. The TV accessory shows up just fine in the Homebridge UI, but is not in the Home app. Other Homebridge accessories appear in the Home app. Configuring the plugin to use switch accessories works fine, but the TV accessory is not working.

Nothing shows up in Home app:
```
        {
            "name": "harmony",
            "hubIP": "192.168.1.229",
            "hubName": "Harmony Hub",
            "DELAY_BEFORE_RETRY_AFTER_NETWORK_LOSS": 60000,
            "HUB_CONNECT_TIMEOUT": 10000,
            "HUB_SEND_TIMEOUT": 30000,
            "TVAccessory": true,
            "sortInput": 0,
            "mainActivity": "Watch Living Room Apple TV",
            "switchAccessories": false,
            "otherPlatforms": [
                {
                    "sortInput": 0
                }
            ],
            "platform": "HarmonyHubWebSocket"
        }
```

Switches show up in Home app:
```
        {
            "name": "harmony",
            "hubIP": "192.168.1.229",
            "hubName": "Harmony Hub",
            "DELAY_BEFORE_RETRY_AFTER_NETWORK_LOSS": 60000,
            "HUB_CONNECT_TIMEOUT": 10000,
            "HUB_SEND_TIMEOUT": 30000,
            "TVAccessory": true,
            "sortInput": 0,
            "mainActivity": "Watch Living Room Apple TV",
            "switchAccessories": true,
            "otherPlatforms": [
                {
                    "sortInput": 0
                }
            ],
            "platform": "HarmonyHubWebSocket"
        }
```

Another interesting thing is that I'm _not_ seeing this issue on Raspbian... I use a separate Homebridge install on my Mac that I use to test changes to things before applying those changes on my Pi. I'm also seeing #403 only on my Mac as well, so maybe the two issues are related?

**Comment by scottleestrange** [comment]
Do this and reset your envorinment


1) Remove the HUB from Home
2) Set cleanCache:true
3) Restart plugin
3) Set cleanCache:false
4) Restart plugin
5) Add back to home.  

My advice - Do not run home bridge form your MAC.  Purchase a cheap Raspberry PI or HOOBS.  It is much easier to isolate issues since it will be dedicated

**Comment by nicoduj** [comment]
Hi, from the description you both made, it seems you did not add the tv accessory to home app manually, but only the bridge . Tv accessory is an external accessory, you have to add it manually to home app (same setup code as the hub itself) . See readme for more detail and let me know .

**Comment by grelca** [comment]
> Hi, from the description you both made, it seems you did not add the tv accessory to home app manually, but only the bridge . Tv accessory is an external accessory, you have to add it manually to home app (same setup code as the hub itself) . See readme for more detail and let me know .

This fixed the issue for me. I had actually tried it already because I noticed the log message about manually adding it, but think I misunderstood how to do that before.. I had just tried scanning the QR code again since it's the same code, but that just represented the bridge and said it had already been added. Needed to click "more options" in order to be able to add the TV.

Interestingly though I didn't need to take that step on my Pi, only on my Mac. On my Pi the TV accessory simply shows up as a device under the bridge.
